wsmatrix@martin.zurowietz.de/overview/thumbnailsBox.js

```diff
@@ -0,0 +1,351 @@
+                    } else {
+                        x += thumbnailWidth - Math.round(thumbnailWidth * thumbnail.collapse_fraction);
+                    }
+                }
+
+                childBox.y1 = box.y1 + thumbnailHeight * targetRow;
+                childBox.y2 = childBox.y1 + thumbnailHeight;
+
+                const indicatorX1 = indicatorLowerX1 +
+                    (indicatorUpperX1 - indicatorLowerX1) * (indicatorValue % 1);
+                const indicatorX2 = indicatorLowerX2 +
+                    (indicatorUpperX2 - indicatorLowerX2) * (indicatorValue % 1);
+                const indicatorY1 = indicatorLowerY1 +
+                    (indicatorUpperY1 - indicatorLowerY1) * (indicatorValue % 1);
+                const indicatorY2 = indicatorLowerY2 +
+                    (indicatorUpperY2 - indicatorLowerY2) * (indicatorValue % 1);
+
+                childBox.x1 = indicatorX1 - indicatorLeftFullBorder;
+                childBox.x2 = indicatorX2 + indicatorRightFullBorder;
+                childBox.y1 = indicatorY1 - indicatorTopFullBorder;
+                childBox.y2 = indicatorY2 + indicatorBottomFullBorder;
+                this._indicator.allocate(childBox);
+            },
+
+            _computeWorkspacesBoxForState(state, workAreaBox, searchHeight, dashHeight, thumbnailsHeight) {
+                const workspaceBox = workAreaBox.copy();
+                const [startX, startY] = workAreaBox.get_origin();
+                const [width, height] = workspaceBox.get_size();
+                const {spacing} = this;
+                const {expandFraction} = this._workspacesThumbnails;
+                let workspaceManager = global.workspace_manager;
+                let rows = workspaceManager.layout_rows;
+
+                switch (state) {
+                    case ControlsState.HIDDEN:
+                        break;
+                    case ControlsState.WINDOW_PICKER:
+                        workspaceBox.set_origin(startX,
+                            startY + searchHeight + spacing +
+                            thumbnailsHeight * rows + spacing * expandFraction);
+                        workspaceBox.set_size(width,
+                            height -
+                            dashHeight - spacing -
+                            searchHeight - spacing -
+                            thumbnailsHeight * rows - spacing * expandFraction);
+                        break;
+                    case ControlsState.APP_GRID:
+                        workspaceBox.set_origin(startX, startY + searchHeight + spacing+
+                            thumbnailsHeight * rows + spacing * expandFraction);
+                        workspaceBox.set_size(
+                            width,
+                            Math.round(height * rows * SMALL_WORKSPACE_RATIO));
+                        break;
+                }
+
+                return workspaceBox;
+            },
+        }
```

Comment:
Is this still here by accident?

Reply:
removing it doesn't seem to change anything, but there was some modifications in the height spacing. I will need to check this more

Reply:
I think this may be a leftover of copy&paste from `controlsManagerLayout.js`. I couldn't find `_computeWorkspacesBoxForState` in the original `thumbnailsBox`.

Reply:
yup, I think you're right, I will fix this now.